test/boost/cql_query_test.cc

```diff
@@ -5063,3 +5063,60 @@ SEASTAR_TEST_CASE(test_parallelized_select_count) {
         BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
     });
 }
+
+SEASTAR_TEST_CASE(test_parallelized_select_min) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, PRIMARY KEY (k));").get();
+        int value_count = 10;
+        for (int i = 0; i < value_count; i++) {
+            e.execute_cql(format("INSERT INTO tbl (k) VALUES ({:d});", i)).get();
+        }
+        auto msg = e.execute_cql("SELECT MIN(k) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {long_type->decompose(int64_t(0))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
+    });
+}
+
+SEASTAR_TEST_CASE(test_parallelized_select_max) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, PRIMARY KEY (k));").get();
+        int value_count = 10;
+        for (int i = 0; i < value_count; i++) {
+            e.execute_cql(format("INSERT INTO tbl (k) VALUES ({:d});", i)).get();
+        }
+        auto msg = e.execute_cql("SELECT MAX(k) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {long_type->decompose(int64_t(value_count - 1))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
+    });
+}
+
+SEASTAR_TEST_CASE(test_parallelized_select_sum) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, PRIMARY KEY (k));").get();
+        int value_count = 10;
+        for (int i = 0; i < value_count; i++) {
+            e.execute_cql(format("INSERT INTO tbl (k) VALUES ({:d});", i)).get();
+        }
+        auto msg = e.execute_cql("SELECT SUM(k) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {long_type->decompose(int64_t((value_count - 1) * value_count / 2))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
```

Comment:
Need some negative checks when we expect a query not to be parallelized.

Reply:
added